crowbar_framework/app/models/api/node.rb

```diff
@@ -150,9 +150,13 @@ def save_node_state(role, state = "upgrading")
       )
       if state == "upgraded"
         progress = status.progress
-        remaining = progress[:remaining_nodes] - 1
-        upgraded = progress[:upgraded_nodes] + 1
-        ::Crowbar::UpgradeStatus.new.save_nodes(upgraded, remaining)
+        # This should not really happen, but in some corner cases,
+        # repeated upgrade of the node could have been invoked
+        if progress[:remaining_nodes] > 0
+          remaining = progress[:remaining_nodes] - 1
+          upgraded = progress[:upgraded_nodes] + 1
+          ::Crowbar::UpgradeStatus.new.save_nodes(upgraded, remaining)
+        end
```

Comment:
Strictly speaking this doesn't seem to belong to this commit. In the future please create separate commits of unrelated changes.